src/modules/ackermann_drive/AckermannDriveGuidance/AckermannDriveGuidance.cpp

```diff
@@ -0,0 +1,272 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "AckermannDriveGuidance.hpp"
+
+#include <mathlib/math/Limits.hpp>
+
+using namespace matrix;
+
+AckermannDriveGuidance::AckermannDriveGuidance(ModuleParams *parent) : ModuleParams(parent)
+{
+	updateParams();
+}
+
+void AckermannDriveGuidance::purePursuit()
+{
+	// uORB subscriber updates
+	if (_position_setpoint_triplet_sub.updated()) {
+		_prev_acc_rad = _position_controller_status.acceptance_radius;
+		_position_setpoint_triplet_sub.copy(&_position_setpoint_triplet);
+	}
+
+	if (_position_controller_status_sub.updated()) {
+		_position_controller_status_sub.copy(&_position_controller_status);
+	}
+
+	if (_vehicle_global_position_sub.updated()) {
+		_vehicle_global_position_sub.copy(&_vehicle_global_position);
+	}
+
+	if (_vehicle_attitude_sub.updated()) {
+		if (_vehicle_attitude_sub.copy(&_vehicle_attitude)) {
+			matrix::Quatf _vehicle_attitude_quaternion = Quatf(_vehicle_attitude.q);
+			_vehicle_yaw = matrix::Eulerf(_vehicle_attitude_quaternion).psi();
+		}
+	}
+
+	if (_home_position_sub.updated()) {
+		_home_position_sub.copy(&_home_position);
+	}
+
+	if (_local_position_sub.updated()) {
+		_local_position_sub.copy(&_local_position);
+	}
+
+	if (_mission_result_sub.updated()) {
+		_mission_result_sub.copy(&_mission_result);
+	}
+
+	if (!_global_local_proj_ref.isInitialized()
+	    || (_global_local_proj_ref.getProjectionReferenceTimestamp() != _local_position.ref_timestamp)) {
+		_global_local_proj_ref.initReference(_local_position.ref_lat, _local_position.ref_lon, _local_position.ref_timestamp);
+	}
+
+	// Setup global frame variables
+	_curr_pos = {_vehicle_global_position.lat, _vehicle_global_position.lon};
```

Comment:
```suggestion
	_curr_pos = Vector2d(_vehicle_global_position.lat, _vehicle_global_position.lon);
```

Reply:
Is this a more efficient way to do this or simply a convention?